doc/langref.html.in

```diff
@@ -6801,6 +6801,32 @@ test "peer type resolution: *const T and ?*T" {
     try expect(a == b);
     try expect(b == a);
 }
+
+test "peer type resolution: error union switch" {
+    // The non-error and error cases are only peers if the error case is just a switch expression;
+    // the pattern `if (x) {...} else |err| blk: { switch (err) {...} }` does not consider the
+    // non-error and error case to be peers.
```

Comment:
I don't like how this turns a limitation of the compiler into a feature of the language.

Reply:
I was just trying to be clear about what the behaviour is so as to not cause confusion, but I can see that this may the source of future problems if peer type resolution changes for these cases. Should I remove the sentence, or add a note saying something along the lines of 'this is the current behaviour and may change'?

Reply:
I think it's good enough for now. I need to go over the langref anyway; it's collected quite a few different contributors' conflicting ideas about what a langref should be and it needs to be reworked by one person with a vision.